tmc-plugin/src/fi/helsinki/cs/tmc/runners/AntExerciseRunner.java

```diff
@@ -25,50 +27,67 @@
+import org.openide.util.Exceptions;
 import org.openide.windows.InputOutput;
 
 public class AntExerciseRunner extends AbstractJavaExerciseRunner {
-    private static final Logger log = Logger.getLogger(AntExerciseRunner.class.getName());
 
-    @Override
-    public Callable<Integer> getCompilingTask(TmcProjectInfo projectInfo) {
-        Project project = projectInfo.getProject();
-        FileObject buildScript = project.getProjectDirectory().getFileObject("build.xml");
-        if (buildScript == null) {
-            throw new RuntimeException("Project has no build.xml");
-        }
-        ExecutorTask task;
-        try {
-            task = ActionUtils.runTarget(buildScript, new String[]{"compile-test"}, null);
-            return executorTaskToCallable(task);
-        } catch (IOException ex) {
-            throw ExceptionUtils.toRuntimeException(ex);
-        }
-    }
+    private static final Logger log = Logger.getLogger(AntExerciseRunner.class.getName());
 
     @Override
-    public Callable<TestRunResult> getTestRunningTask(final TmcProjectInfo projectInfo) {
-        return new Callable<TestRunResult>() {
+    public Callable<Optional<TestRunResult>> getTestRunningTask(final TmcProjectInfo projectInfo) {
+        return new Callable<Optional<TestRunResult>>() {
             @Override
-            public TestRunResult call() throws UserVisibleException, IOException, InterruptedException, ExecutionException {
-                FileObject testDir = findTestDir(projectInfo);
-                if (testDir == null) {
-                    throw new UserVisibleException("No test directory in project");
-                }
+            public Optional<TestRunResult> call() throws Exception {
 
-                List<TestMethod> tests = findProjectTests(projectInfo, testDir);
+                log.log(INFO, "Starting compile");
+
+                Project project = projectInfo.getProject();
+                FileObject buildScript = project.getProjectDirectory().getFileObject("build.xml");
+                if (buildScript == null) {
+                    throw new RuntimeException("Project has no build.xml");
+                }
+                ExecutorTask task;
 
-                File tempFile;
-                tempFile = File.createTempFile("tmc_test_results", ".txt");
                 try {
-                    return runTests(projectInfo, testDir, tests, tempFile);
-                } finally {
-                    tempFile.delete();
+                    task = ActionUtils.runTarget(buildScript, new String[]{"compile-test"}, null);
+                } catch (IOException ex) {
+                    Exceptions.printStackTrace(ex);
+                    throw ExceptionUtils.toRuntimeException(ex);
+                } catch (IllegalArgumentException ex) {
+                    Exceptions.printStackTrace(ex);
+                    throw ExceptionUtils.toRuntimeException(ex);
+                }
+
+                int compileResult = task.result();
```

Comment:
Might be neat to factor the compiling into a helper method like you did with testing.